packages/react-component-library/src/components/DataGrid/DataGrid.tsx

```diff
@@ -177,12 +205,35 @@ export const DataGrid = <T extends object>(props: DataGridProps<T>) => {
 
   const handlePagination = useCallback(
     (...args) => {
-      const [_, currentPage] = args
-      table.setPageIndex(currentPage - 1)
-      // @ts-ignore
-      onPageChange?.(...args)
+      const [event, currentPage, totalPages] = args
+      const pageIndex = currentPage - 1
+
+      // For external pagination, use onPaginationChange
```

Comment:
I sometimes wonder if we need comments then is the prop name right. Is `manualPagination` the right name?

Reply:
Yeah so that prop is actually typed by the underlying library (we also have `manualSorting`).

I'd agree the comments are probably unnecessary - I'll remove them.

![Screenshot 2025-04-16 at 08 39 23](https://github.com/user-attachments/assets/c1e53798-0822-459f-9cdd-15248b428d67)

Reply:
I'll probably just leave this one:

```
// Legacy support for onPageChange
```